src/Session.cc

```diff
@@ -516,7 +518,7 @@ bool Session::make_private_shared(AutoRemoteSyscalls& remote,
 
   // Now create the new mapping in its place
   const AddressSpace::Mapping& new_m = remote.task()->vm()->mapping_of(
-      create_shared_mmap(remote2, sz, start).start());
+      create_shared_mmap(remote2, sz, start, m.map.prot()).start());
```

Comment:
I merged my fix into your commit so I don't have it handy, but the bug is here. The `remap()` call above may have invalidated `m`. I fixed it by making the `m` parameter a value instead of a reference.